### DISCUSSION TITLE
Delay client routing (until transition is complete)

### DISCUSSION THREAD

**Discussion by lutzissler** [vikejs/vike#633] [Discussion]
First, thanks for building vite-plugin-ssr! It’s a relief not having to jump on that big Nuxt.js tanker for getting some SSR done.

I am building a Vue app/website and using `clientRouting = true`. My app is using page transitions which take a (small) while. Half of the transition/animation should happen on the old page, then the navigation should occur, and then the other half of the transition should happen on the new page. Hence what is needed is a way to _delay_ the actual routing.

In vue-router, this can be done with a ‘global before guard’ (see https://router.vuejs.org/guide/advanced/navigation-guards.html#global-before-guards), which can be an `async` function or allows to return a Promise. If used that way, the navigation will only occur when the Promise resolves to `true`. In other words, I can simply do this:

```
router.beforeEach((to, from) => {
    return new Promise(resolve => {
      setTimeout(resolve, 150)
    })
})
```

…and the actual routing will be delayed by 150ms.

Is there any way to achieve this with client routing in vite-plugin-ssr?

**Comment by brillout** [comment]
You can't delay the rendering process *start*, but you can delay the actual Vue rendering (just sleep in your client-side `render()` hook).

If you can create a convincing UX story around it, I may consider a feature request.

**Reply by lutzissler** [reply]
Imagine a website which, when any link is clicked, shows an effect to hide the current page (like a theater curtain), then goes to the next page (behind that curtain), then pulls up the curtain again. We would need to delay the actual routing here (with a Promise) because we want to make sure that curtain is fully before navigating off the original page.

In other words, I don’t want to delay the _rendering process_ start. I want to delay the _routing_ start.

**Reply by brillout** [reply]
I see. So, yea, as proposed in my previous comment I think you can achieve what you want without delaying routing start, but instead by delaying the Vue rendering in `render()`.

**Reply by lutzissler** [reply]
Thanks again for your response. Is that true? I think I will have to ‘drop the curtain’ _before_ the routing starts because otherwise the old page will be gone before the curtain has dropped. Or am I missing something?

**Reply by brillout** [reply]
You can achieve that.

**Reply by lutzissler** [reply]
Ok, I suppose you mean by delaying the `changePage()` call in `render()`, like so:

```
setTimeout(() => {
    app.changePage(pageContext)
}, 150)
```

Thanks.
